kahuna/public/js/util/async.js

```diff
@@ -68,6 +68,10 @@ async.factory("apiPoll", [
       if (status === 'fulfilled') {
         return $q.resolve(value);
       }
+      // Something has gone wrong, so we can let the user know
+      if (n > 100) {
+        throw new Error('gave up after 100 tries (apiPoll failed)');
```

Comment:
This is a compromise text with the user-friendly text followed by some technical info for debugging.